Add more information about epics-base documentation workflow (#808)
epics-base/epics-base: documentation/README.md

### Documentation

The general EPICS documentation is available from the
[EPICS Documentation Website](https://docs.epics-controls.org/).
This is based on the [EPICS Docs Github project](https://github.com/epics-docs/epics-docs).

The documentation for EPICS base is included into the general documentation,
but is specifically available from the [EPICS Documentation Base Project](https://docs.epics-controls.org/projects/base/).

The source files for EPICS base documentation can be found in the
`base/documentation` directory of the distribution.

#### Documentation Build Pipeline

The documentation of EPICS base is automatically built via ReadTheDocs.

> The configuration is defined in `base/documentation/.readthedocs.yaml`.

Reviewer: This is not the correct location for this file, it is at the top level of the repo.

Author: good catch! thanks

> Any commit triggers a webhook for the [EPICS Base ReadTheDocs project](https://app.readthedocs.org/projects/epics-base/).

Reviewer 2: Not completely true, it's only for commit on the main branch, and opening pull requests (and specifically adding RtD versions on other tags/branches). The WebHook mechanism is also going away in the future, in favor of a GitHub OAuth2 App.

```suggestion
Any commit on the main branch, and opening pull requests triggers a build of the [EPICS Base ReadTheDocs project](https://app.readthedocs.org/projects/epics-base/).
```

> More information on how to locally build the documentation can be found in the [building the documentation]({doc}`build-docs`) section.

Reviewer 2: I don't know why I didn't catch that last time, but I don't think this syntax works, it should be:

```suggestion
More information on how to locally build the documentation can be found in the {doc}`build-docs` section.
```

Which should work once the other PR is merged.

Author: fixed

> More information on how to locally build the documentation can be found in the [building the documentation](:doc:`build-docs`) section.

Reviewer 2: ```suggestion
More information on how to locally build the documentation can be found in the [building the documentation]({doc}`build-docs`) section.
```

### Documentation

The general EPICS documentation is available from the
[EPICS Documentation Website](https://docs.epics-controls.org/).
This is based on the [EPICS Docs Github project](https://github.com/epics-docs/epics-docs).

The documentation for EPICS base is included into the general documentation,
but is specifically available from the [EPICS Documentation Base Project](https://docs.epics-controls.org/projects/base/).

The source files for EPICS base documentation can be found in the
`base/documentation` directory of the distribution.

#### Documentation Build Pipeline

> The documentation of EPICS base is automatically build via ReadTheDocs.

Reviewer 3: s/build/built/ please.

The configuration is defined in `base/documentation/.readthedocs.yaml`.
Any commit triggers a webhook for the [EPICS Base ReadTheDocs project](https://app.readthedocs.org/projects/epics-base/).

#### Manually Building EPICS Base documentation

To build the EPICS Base documentation the requirements are:

- [Doxygen](https://www.doxygen.nl/index.html) (Suggested version 1.8)

> - Python modules as defined in `base/documentation/requirements.txt`

Reviewer 3: NB: @minijackson is changing this...

These requirements reflect the versions which are available in the build image used by the ReadTheDocs pipeline.
There is a known issue with using newer versions of Doxygen,
leading to issues with the Sphinx Breathe extension ([as discussed in #783](https://github.com/epics-base/epics-base/issues/783)).

To locally build the documentation, first run:

```bash
make inc
```

This prepares the header files and parses the `.pod` files for the component reference documentation.

Then navigate to the `documentation/` folder and run `make sphinx`:

```bash
cd documentation/
make sphinx
```

This invokes doxygen and certain scripts to prepare the Sphinx build,
and also triggers the Sphinx build itself.

> Using `make sphinx DOX=`, the doxygen step can be skipped.

Reviewer 3: Add "to speed up the build after a minor local update" here?

Author: removed the section
